### DISCUSSION TITLE
This example in the docs seems a bit odd, and a basic question

### DISCUSSION THREAD

**Discussion by JohnAtl** [treeverse/dvc#9263] [Help] [answered]
(Tried this on the Discourse, but discussions seem to be happening here.)

I’m a bit of a noob, and I’m trying to understand how dvc install changes the steps introduced in Dr. O’Brien’s introductory videos.

Question 1

The final example on [this page](https://dvc.org/doc/command-reference/install) of the docs shows a source file stored in a dvc repository, thus when changed, dictates running repro to create the data, then everything is somehow up to date.

I would think that:

source would be saved to git
running modified source with repro would result in a changed dataset that would need to be added, committed, and pushed.

Question 2

When I’m working on my project and I run git commit, dvc status is run and (let’s assume) shows me that some of my dvc-managed files have changed. Am I correct in these commands (from memory):

\# add all modified files in data folder
dvc add data
dvc push
\# make sure files changed by dvc makes sense
git status 
git add -a -m “dvc modified”
git push
This seems pretty convoluted, so if there’s a better way, would love to know. If not, not looking a gift horse in the mouth :slightly_smiling_face:

Question 3

Okay, a bonus - what about merging, say, a branch into main? Say, the signatures in the *.dvc files don’t match. Is it just a matter of always selecting the branch data over main in the conflicted .dvc files (unless something has gone really wrong)?

Thanks!

**Comment by dberenbaum** [comment, marked as answer]
Hi @JohnAtl, it sounds like you have the right ideas here.

**Question 1**

This example is based on the [example pipeline in our get started docs](https://dvc.org/doc/start/data-management/data-pipelines). `src/featurization.py` is tracked in git, but it's also a dependency of a stage in the pipeline, so dvc still checks whether the pipeline has been run with the changes to that file. The changed dataset is tracked in dvc by `dvc.lock`, which is what you see being added in that example.

**Question 2**

Yup, that looks about right. It is very explicit, but we prefer to make it that way so it's clear what's happening.

**Question 3**

Have you taken a look at https://dvc.org/doc/user-guide/how-to/resolve-merge-conflicts?

**Reply by JohnAtl** [reply]
Thanks for the information!
Sometimes it's good just to know what 'normal' is.
